vendor/k8s.io/kubectl/pkg/util/rbac/rbac.go

```diff
@@ -42,7 +42,11 @@ func CompactRules(rules []rbacv1.PolicyRule) ([]rbacv1.PolicyRule, error) {
 				if existingRule.Verbs == nil {
 					existingRule.Verbs = []string{}
 				}
-				existingRule.Verbs = append(existingRule.Verbs, rule.Verbs...)
+				for _, v := range rule.Verbs {
```

Comment:
I'd suggest using `k8s.io/apimachinery/pkg/util/sets#String` here, so you don't have to deal with duplicates manually.
Just create a `sets.String` and add verbs to it, after loop you'll assign it back to `existingRule.Verbs`

Reply:
thanks!